From dc5c6673b8dc472fdbac0f162f722b67a1f18ad6 Mon Sep 17 00:00:00 2001
From: Timothy E Baldwin <T.E.Baldwin99@members.leeds.ac.uk>
Date: Thu, 23 Feb 2017 22:36:58 +0000
Subject: [PATCH 3/5] Srcbuild: Remove uneeded sb_system_sync() calls.

---
 castle/RiscOS/Utilities/Release/srcbuild/c/build | 30 ------------------------
 1 file changed, 30 deletions(-)

diff --git a/castle/RiscOS/Utilities/Release/srcbuild/c/build b/castle/RiscOS/Utilities/Release/srcbuild/c/build
index 603de7b5..1682a1ae 100644
--- a/castle/RiscOS/Utilities/Release/srcbuild/c/build
+++ b/castle/RiscOS/Utilities/Release/srcbuild/c/build
@@ -438,8 +438,6 @@ int build_clean(void)
     component = component->next;
   }
 
-  sb_system_sync();
-
   return 0;
 }
 
@@ -496,8 +494,6 @@ int build_clean_all(void)
     component = component->next;
   }
 
-  sb_system_sync();
-
   return 0;
 }
 
@@ -512,8 +508,6 @@ static int build_export(const char *phase)
   module_entry_t *module;
   module_type_t type;
 
-  sb_system_sync();
-
   component = component_database;
   while (component) {
     module = lookup_module(component->name);
@@ -555,8 +549,6 @@ static int build_export(const char *phase)
     component = component->next;
   }
 
-  sb_system_sync();
-
   return 0;
 }
 
@@ -611,7 +603,6 @@ static int build_resources_(build_resources_reason_code reason_code)
     else
     {
       sprintf(buffer,"-k clean COMPONENT=Messages TARGET=Messages NUMLOCALE=%s", locale);
-      sb_system_sync();
       if (amu_machinef(buffer) != 0)
         error(0, "running make clean on module 'Messages'.");
     }
@@ -627,8 +618,6 @@ static int build_resources_(build_resources_reason_code reason_code)
     build_dirs();
   }
 
-  sb_system_sync();
-
   /*
    * now get on with rest
    */
@@ -704,8 +693,6 @@ static int build_resources_(build_resources_reason_code reason_code)
     component = component->next;
   }
 
-  sb_system_sync();
-
   /* Finally change the locale back to the default one (the first in the list). */
   build_select_locale(supported_territories[0]);
 
@@ -1008,7 +995,6 @@ int build_bbe(void)
 
           }
         }
-        sb_system_sync();
         strcpy(end_rom_link_name, "_v");
         rom_link_file = fopen(rom_link_name, "r");
         if (rom_link_file)
@@ -1286,8 +1272,6 @@ build_install_rom_(build_install_rom_reason_code reason_code)
   char scriptfilename[BUFFER_LEN];
   char forcefilename[BUFFER_LEN];
 
-  sb_system_sync();
-
   if (reason_code == build_install_rom_bbe_script)
   {
     FILE *f;
@@ -1392,7 +1376,6 @@ build_install_rom_(build_install_rom_reason_code reason_code)
           }
 
           /* Need to stat the module file, display position in ROM */
-          sb_system_sync();
           size = build_get_filesize(output_path, module->filename);
 
 	  /* keep record of things for later */
@@ -1447,8 +1430,6 @@ build_install_rom_(build_install_rom_reason_code reason_code)
     ran_install_rom = (reason_code == build_install_rom_normal);
   }
 
-  sb_system_sync();
-
   return 0;
 }
 
@@ -1838,9 +1819,7 @@ build_join(void)
 
 
   /* dispatch the joiner */
-  sb_system_sync();
   system_result = sb_system(buffer);
-  sb_system_sync();
   remove(filename);
 
   if (system_result != 0) {
@@ -1943,8 +1922,6 @@ build_install(void)
   if ((fh_shadow = fopen(buffer, "w")) == NULL)
     error(1, "cannot open shadow file '%s'.", buffer);
 
-  sb_system_sync();
-
   component = component_database;
   while (component) {
     module = lookup_module(component->name);
@@ -1971,7 +1948,6 @@ build_install(void)
 		component->name,
 		component->options ? component->options : "") != 0)
 	      error(0, "running make clean on module '%s'.", component->name);
-              sb_system_sync();
 	  }
 	  /*
 	   * find out what we are switching this module on
@@ -2034,8 +2010,6 @@ build_install(void)
   if (fh_shadow)
     fclose(fh_shadow);
 
-  sb_system_sync();
-
   return 0;
 }
 
@@ -2060,8 +2034,6 @@ build_release_autobuild(void)
   if ((fh_shadow = fopen(buffer, "w")) == NULL)
     error(1, "cannot open shadow file '%s'.", buffer);
 
-  sb_system_sync();
-
   component = component_database;
   while (component) {
     module = lookup_module(component->name);
@@ -2140,8 +2112,6 @@ build_release_autobuild(void)
   if (fh_shadow)
     fclose(fh_shadow);
 
-  sb_system_sync();
-
   return 0;
 }
 
-- 
2.11.0

