From c2cf0f7ab8a1215ecc5afb683e7ae802dcb65cab Mon Sep 17 00:00:00 2001
From: Timothy E Baldwin <T.E.Baldwin99@members.leeds.ac.uk>
Date: Thu, 23 Aug 2018 21:47:34 +0100
Subject: [PATCH] linux-user: SWI Intercept and disable FPA (for RISC OS)

Signed-off-by: Timothy E Baldwin <T.E.Baldwin99@members.leeds.ac.uk>
---
 linux-user/arm/cpu_loop.c | 16 ++++++++++++++--
 1 file changed, 14 insertions(+), 2 deletions(-)

diff --git a/linux-user/arm/cpu_loop.c b/linux-user/arm/cpu_loop.c
index 26928fbbb2..ac6f96845e 100644
--- a/linux-user/arm/cpu_loop.c
+++ b/linux-user/arm/cpu_loop.c
@@ -206,6 +206,10 @@ do_kernel_trap(CPUARMState *env)
 
 void cpu_loop(CPUARMState *env)
 {
+    static bool use_nwfpe = true;
+    static abi_ulong sys_base = 0;
+    static abi_ulong sys_limit = 0xFFFFFFFF;
+
     CPUState *cs = CPU(arm_env_get_cpu(env));
     int trapnr;
     unsigned int n, insn;
@@ -233,7 +237,7 @@ void cpu_loop(CPUARMState *env)
                 /* FIXME - what to do if get_user() fails? */
                 get_user_code_u32(opcode, env->regs[15], env);
 
-                rc = EmulateAll(opcode, &ts->fpa, env);
+                rc = use_nwfpe && EmulateAll(opcode, &ts->fpa, env);
                 if (rc == 0) { /* illegal instruction */
                     info.si_signo = TARGET_SIGILL;
                     info.si_errno = 0;
@@ -322,7 +326,10 @@ void cpu_loop(CPUARMState *env)
                     }
                 }
 
-                if (n == ARM_NR_cacheflush) {
+                if (env->regs[15] < sys_base || env->regs[15] > sys_limit) {
+                    info.si_signo = TARGET_SIGSYS;
+                    queue_signal(env, info.si_signo, QEMU_SI_FAULT, &info);
+                } else if (n == ARM_NR_cacheflush) {
                     /* nop */
                 } else if (n == ARM_NR_semihosting
                            || n == ARM_NR_thumb_semihosting) {
@@ -372,6 +379,11 @@ void cpu_loop(CPUARMState *env)
                             env->regs[0] = ret;
                         }
                     }
+                } else if (n == 0xC0001) {
+                    /* RISC OS: Control SWI Intercept and FPA */
+                    sys_base = env->regs[0];
+                    sys_limit = env->regs[1];
+                    use_nwfpe = env->regs[2];
                 } else {
                     goto error;
                 }
-- 
2.18.0

