diff -ur a/src/arm_common.c b/src/arm_common.c
--- a/src/arm_common.c	2016-10-26 17:51:41.000000000 +0100
+++ b/src/arm_common.c	2018-01-12 13:14:31.991567734 +0000
@@ -20,6 +20,8 @@
 #define SWI_OS_CallASWI		0x6f
 #define SWI_OS_CallASWIR12	0x71

+#define SWI_IXSupport_LinuxSyscall	0x59EC0
+
 #define SWI_Portable_ReadFeatures	0x42fc5
 #define SWI_Portable_Idle		0x42fc6

@@ -375,6 +377,10 @@
		mouse_hack_osmouse();
		arm.reg[cpsr] &= ~VFLAG;

+	} else if (swinum == SWI_IXSupport_LinuxSyscall && (arm.reg[7] == 1 || arm.reg[7] == 248)) {
+		/* IXSupport_LinuxSyscall exit */
+		exit(arm.reg[0]);
+
	} else if (swinum == ARCEM_SWI_HOSTFS) {
		ARMul_State state;
		state.Reg = arm.reg;
